src/lib/hosts-file.ts

```diff
@@ -0,0 +1,175 @@
+import fs from 'fs';
+import { platform, tmpdir } from 'os';
+import path from 'path';
+import { promisify } from 'util';
+import sudo from 'sudo-prompt';
+
+const readFile = promisify( fs.readFile );
+const writeFile = promisify( fs.writeFile );
+const sudoExec = promisify( sudo.exec );
+
+// Host file paths for different operating systems
+const HOST_FILES: Record< string, string > = {
+	win32: 'C:\\Windows\\System32\\drivers\\etc\\hosts',
```

Comment:
```suggestion
	win32: path.resolve( process.env.SystemRoot ?? 'C:\\Windows', 'System32\\drivers\\etc\\hosts' ),
```
Nit, but this seems appropriate.